62.CAD/Hatch.cpp

```diff
@@ -0,0 +1,782 @@
+                {
+                    for (uint32_t itemIdx = section.index; itemIdx < section.index + section.count; itemIdx ++)
+                    {
+                        auto lineBezier = line.getQuadBezierInfoAt(itemIdx);
+                        QuadraticBezier unsplitBezier = { { lineBezier.p[0], lineBezier.p[1], lineBezier.p[2] }};
+                        
+                        // Beziers must be monotonically increasing along major
+                        // First step: Make sure the bezier is monotonic, split it if not
+                        std::array<QuadraticBezier, 2> monotonicSegments;
+                        auto isMonotonic = unsplitBezier.splitIntoMajorMonotonicSegments(monotonicSegments);
+
+                        if (isMonotonic)
+                        {
+                            // Already was monotonic
+                            addBezier(unsplitBezier);
+                            if (debugOutput)
+                                drawDebugBezier(unsplitBezier, float32_t4(0.8, 0.8, 0.8, 0.2));
+                        }
+                        else
+                        {
+                            addBezier(monotonicSegments.data()[0]);
+                            addBezier(monotonicSegments.data()[1]);
+                            if (debugOutput)
+                            {
+                                drawDebugBezier(monotonicSegments.data()[0], float32_t4(0.0, 0.6, 0.0, 0.5));
+                                drawDebugBezier(monotonicSegments.data()[1], float32_t4(0.0, 0.0, 0.6, 0.5));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        for (uint32_t bezierIdx = 0; bezierIdx < beziers.size(); bezierIdx++)
+        {
+            auto hatchBezier = &beziers[bezierIdx];
+            Segment segment;
+            segment.originalBezier = hatchBezier;
+            segment.t_start = 0.0;
+            segment.t_end = 1.0;
+            segments.push_back(segment);
+        }
+
+        // TODO better way to do this
+        std::sort(segments.begin(), segments.end(), [&](Segment a, Segment b) { return a.originalBezier->p[0][major] > b.originalBezier->p[0][major]; });
+		for (Segment& segment : segments)
+		{
+			starts.push(segment);
+			//std::cout << "Starts: " << segment.originalBezier->p[0][major]  << "\n";
+		}
+
+        std::sort(segments.begin(), segments.end(), [&](Segment a, Segment b) { return a.originalBezier->p[2][major] > b.originalBezier->p[2][major]; });
+        for (Segment& segment : segments)
+		{
+			ends.push(segment.originalBezier->p[2][major]);
+			//std::cout << "Ends: " << segment.originalBezier->p[2][major] << "\n";
+		}
+        maxMajor = segments.front().originalBezier->p[2][major];
+    }
+
+    // Sweep line algorithm
+	std::priority_queue<double, std::vector<double>, std::greater<double> > intersections; // Next intersection points as major coordinate
```

Comment:
move this line to above next to start and end, makes more sense in the context for readability